sdk/python/lib/pulumi/runtime/stack.py

```diff
@@ -47,10 +43,11 @@
         #
         # Note that "asyncio.sleep(0)" is the blessed way to do this:
         # https://github.com/python/asyncio/issues/284#issuecomment-154180935
-        await asyncio.sleep(0)
-
-        # Wait for all outstanding RPCs to retire.
-        await RPC_MANAGER.wait_for_outstanding_rpcs()
+        while True:
+            await asyncio.sleep(0)
+            if RPC_MANAGER.count == 0 or RPC_MANAGER.unhandled_exception is not None:
+                log.debug(f"Wait finished: count is {RPC_MANAGER.count}, exn is {RPC_MANAGER.unhandled_exception}")
+                break
```

Comment:
The comment directly after this implies that when we get to L58, all outstanding RPCs are complete.  I see how that was enforced under the old code, but it seems like with the new code, it's possible for us to get to L58 with multiple pending RPCs in flight, if `RPC_MANAGER.unhandled_exception` is not `None`.

Is it okay for us to cancel these pending RPCs in this case?

Reply:
>  it seems like with the new code, it's possible for us to get to L58 with multiple pending RPCs in flight, if RPC_MANAGER.unhandled_exception is not None.

Good catch. I've updated the code to match the original behavior.